### PR TITLE
`Query.Bounds` - add overload for collection of bounding boxes, `DoesIntesect` - add transform input

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #

- Add `Bounds` for `IEnumerable<BoundingBoxXYZ>`
- Add transform input to `DoesIntersect`

Breaking change in DoesIntersect(BoundingBoxXYZ, Element) method - element transform calculation has been removed from the body as it can produce incorrect results. Transform of element has been added as an input.

### CODE REVIEW

File: Revit_Core_Engine/Query/Bounds.cs [hunk 1]
```diff
@@ -150,6 +150,34 @@ public static BoundingBoxXYZ Bounds(this IEnumerable<XYZ> points)
...
+
+            foreach (BoundingBoxXYZ bbox in bboxes)
+            {
+                Transform transform = bbox.Transform;
+
+                if (!transform.IsIdentity)
+                {
+                    points.Add(transform.OfPoint(bbox.Min));
+                    points.Add(transform.OfPoint(bbox.Max));
+                }
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/Bounds.cs:169]: Are you sure this is true? Shouldn't we use `BoundsOfTransformed` here?

Comment (michal-pekacki) [reply Revit_Core_Engine/Query/Bounds.cs:169]: Yes, I will change it 👍

File: Revit_Core_Engine/Query/Bounds.cs [hunk 2]
```diff
@@ -150,6 +150,35 @@ public static BoundingBoxXYZ Bounds(this IEnumerable<XYZ> points)
         }
 
         /***************************************************/
+
+        [Description("Returns combined bounding box of the given collection of bounding boxes.")]
+        [Input("bboxes", "A collection of the bounding boxes to get the bounds for.")]
+        [Output("bounds", "Combined bounding box of the bounding boxes.")]
+        public static BoundingBoxXYZ Bounds(this IEnumerable<BoundingBoxXYZ> bboxes)
+        {
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/Bounds.cs:158]: I think we can achieve this in a single line:
```C#
        return bboxes.SelectMany(x => x.CornerPoints()).ToList().Bounds();
```

Comment (michal-pekacki) [reply Revit_Core_Engine/Query/Bounds.cs:158]: Yes! This is the simplest way 👍

File: Revit_Core_Engine/Query/DoesIntersect.cs [hunk 3]
```diff
@@ -67,15 +67,18 @@ public static bool DoesIntersect(this Element element1, Element element2)
         [Description("Check if bounding box intersects with element.")]
         [Input("bbox", "Bounding box to check the intersection for.")]
         [Input("element", "Element to check the intersection for.")]
+        [Input("transform", "Transform of the element. If element is from revit link, revit link instance transform should be used.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/DoesIntersect.cs:70]:
```suggestion
        [Input("transform", "Transform of the element. In most usual case of a linked element being checked against a bounding box in host document's coordinate system, Revit link instance transform should be used.")]
```

**Review by pawelbaran — DISMISSED** [review]
Happy to approve following a good process with @michal-pekacki 👍

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check core
@BHoMBot check null-handling
@BHoMBot check serialisation

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Comment by michal-pekacki** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check dataset-compliance

**Review by FraserGreenroyd — APPROVED** [review]
Approving based on @pawelbaran review as commit since that review was for versioning additions.

**Pull request merged**